deepspeed/runtime/fp16/onebit_adam.py

```diff
@@ -0,0 +1,395 @@
+'''
+Copyright 2019 The Microsoft DeepSpeed Team
+'''
+import types
+import torch
+import importlib
+import numpy as np
+import time
+import cupy
+from torch.utils.dlpack import to_dlpack
+from torch.utils.dlpack import from_dlpack
+from deepspeed.utils.logging import logger
+
+from mpi4py import MPI
+from deepspeed.runtime.custom_collectives import gather, allgather
+
+
+class OnebitAdam(torch.optim.Optimizer):
+    """Implements the 1-bit Adam algorithm. Currently GPU-only.
+    For usage example please see, TODO DeepSpeed Tutorial
+    It has been proposed in TODO: add paper name and arXiv link.
+
+    Arguments:
+        params (iterable): iterable of parameters to optimize or dicts defining
+            parameter groups.
+        lr (float, optional): learning rate. (default: 1e-3)
+        betas (Tuple[float, float], optional): coefficients used for computing
```

Comment:
Update the docstring here to add details about all the arguments that OnebitAdam uses, e.g., freeze_step is a pretty important one to document here (and other places too).

Reply:
Add the detail.